pkg/meta/model/job_args.go

```diff
@@ -919,34 +919,64 @@ type DropColumnArgs struct {
 	PartitionIDs []int64 `json:"partition_ids,omitempty"`
 }
 
-func (a *DropColumnArgs) fillJobV1(job *Job) {
-	job.Args = []any{a.ColName, a.IfExists, a.IndexIDs, a.PartitionIDs}
+// AddColumnArgs is the arguments for adding column ddl.
+type AddColumnArgs struct {
+	Col         *ColumnInfo         `json:"column_info,omitempty"`
+	Pos         *ast.ColumnPosition `json:"position,omitempty"`
+	Offset      int                 `json:"offset,omitempty"`
+	IfNotExists bool                `json:"if_not_exists,omitempty"`
 }
 
-// GetDropColumnArgs gets the args for drop column ddl.
-func GetDropColumnArgs(job *Job) (*DropColumnArgs, error) {
-	var (
-		colName      pmodel.CIStr
-		ifExists     bool
-		indexIDs     []int64
-		partitionIDs []int64
-	)
+// TableColumnArgs is the arguments for dropping column ddl or Adding column ddl.
+type TableColumnArgs struct {
+	*DropColumnArgs `json:"drop_column_args,omitempty"`
+	*AddColumnArgs  `json:"add_column_args,omitempty"`
```

Comment:
we don't need 3 args types, can merge them into 1

ColName can be stored inside `ColumnInfo`